pkg/orb/orb.go

```diff
@@ -164,7 +179,8 @@ func Reconfigure(ctx context.Context, monitor mntr.Monitor, orbConfig *Orb, newR
 
 	// If the repokey already has read/write permissions, don't generate a new one.
 	// This ensures git providers other than github keep being supported
-	if err := configureGit(false); err != nil {
+	// Only if you're not trying to set a new key, as you don't want to generate a new key then
+	if err := configureGit(false); err != nil && newRepoKey == "" {
```

Comment:
if newRepoKey != "", configureGit(true) should be called directly

Reply:
Changed to only generate ssh-keys if it's github and no specific key is given as parameter.